include/cute/atom/copy_traits_xe.hpp

```diff
@@ -1,80 +1,324 @@
-            int H = size<0>(traits.tensor);
-            // int W = size<1>(traits.tensor) * sizeof(typename decltype(traits.tensor)::engine_type::value_type);
-            int W = size<1>(traits.tensor) * sizeof(typename TD::value_type); //TODO: inconsistent to give the size in elements but use vector for copy
-            auto [y, x] = src.data().coord_;
-            XE_2D_LOAD::copy(traits.tensor.data().get(), W, H, W, int2_{x, y}, &*dst.data());
-        }
-    };
-
-
-    template <class GTensor>
-    struct Copy_Traits<XE_2D_SAVE, GTensor>
-    {
-        // using ThrID   = Layout<_16>; //TODO: I think it should be 16 (copy is per subgroup) - but static_assert fails
-        using ThrID = Layout<_1>;
-        using NumBits = Int<sizeof(typename GTensor::engine_type::value_type) * 8>; // hacky: does vec of 8
-        // Map from (src-thr,src-val) to bit
-        using SrcLayout = Layout<Shape<_1, NumBits>>; // TODO:  is _1 correct?
-        // Map from (dst-thr,dst-val) to bit
-        using DstLayout = Layout<Shape<_1, NumBits>>;
-        // Reference map from (thr,val) to bit
-        using RefLayout = SrcLayout;
-
-        GTensor tensor;
-
-        template <class TS, class SLayout,
-                  class TD, class DLayout>
-        CUTE_HOST_DEVICE friend constexpr void
-        copy_unpack(Copy_Traits const &traits,
-                    Tensor<TS, SLayout> const &src,
-                    Tensor<ViewEngine<ArithmeticTupleIterator<TD>>, DLayout> &dst)
-        {
-            static_assert(is_rmem<TS>::value);
-            int H = size<0>(traits.tensor);
-            int W = size<1>(traits.tensor) * sizeof(typename decltype(traits.tensor)::engine_type::value_type);
-            auto [y, x] = dst.data().coord_;
-            XE_2D_SAVE::copy(traits.tensor.data().get(), W, H, W, int2_{x, y}, &*src.data());
-        }
-    };
-
-    template <class Copy, class GEngine, class GLayout>
-    auto make_xe_2d_copy(Tensor<GEngine, GLayout> gtensor)
-    {
-        using GTensor = Tensor<GEngine, GLayout>;
-        using Traits = Copy_Traits<Copy, GTensor>;
-        Traits traits{gtensor};
-        return Copy_Atom<Traits, typename GEngine::value_type>{traits};
-    }
+template <class CopyOp, class GTensor>
+struct XE_2D_LD_Unpack
+{
+  GTensor tensor;
+
+  using Copy_Traits = Copy_Traits<CopyOp, GTensor>;
+  template <class TS, class SLayout,
+            class TD, class DLayout>
+  CUTE_HOST_DEVICE friend constexpr void
+  copy_unpack(Copy_Traits const &traits,
+              Tensor<ViewEngine<ArithmeticTupleIterator<TS>>, SLayout> const &src,
+              Tensor<TD, DLayout> &dst)
+  {
+      static_assert(is_rmem<TD>::value);
+      int H = size<0>(traits.tensor);
+      int W = size<1>(traits.tensor) * sizeof(typename Copy_Traits::CopyInternalType);
+      auto [y, x, z] = src.data().coord_;
+      CopyOp::copy(traits.tensor.data() + z, W, H, W, intel::coord_t{static_cast<int>(x), static_cast<int>(y)}, &*dst.data());
```

Comment:
This cast isn't needed for me. I tested it with https://github.com/muhammad-tanvir-1211/cutlass-fork/commit/d138e84c565a0df0b4b23db50b269cb90aec970f and https://github.com/intel/llvm/releases/tag/nightly-2024-04-30 (both with libstdc++12 and libc++ nightly)

Reply:
@rolandschulz  thanks for checking! we have tested this internally with the version you mentioned on our CI side and we can confirm that is not needed as well. We will check with @muhammad-tanvir-1211  to see what is his set up causing issue for him

Reply:
Thank you both. I can confirm that it was a problem with my setup, I was able to get the code working on my system without these `static_casts`. Apologies for the confusion earlier :)